controller/monitor/snapshot_monitor.go

```diff
@@ -306,31 +306,26 @@ func (m *SnapshotMonitor) UpdateConfiguration(map[string]interface{}) error {
 		longhorn.DataEngineTypeV1,
 		longhorn.DataEngineTypeV2,
 	} {
-		if dataIntegrityCronJobs[dataEngine] != m.existingDataIntegrityCronJobs[dataEngine] || m.checkSchedulers[dataEngine].Len() == 0 {
-			if m.checkSchedulers[dataEngine].Len() > 0 {
-				m.checkSchedulers[dataEngine].Remove(m.scheduledJobs[dataEngine])
-
-			}
-			job, err := m.checkSchedulers[dataEngine].Cron(dataIntegrityCronJobs[dataEngine]).Do(func() {
-				m.checkSnapshots(dataEngine)
-			})
+		if m.checkSchedulers[dataEngine].Len() > 0 {
+			m.checkSchedulers[dataEngine].Remove(m.checkSnapshots)
+		}
+		job, err := m.checkSchedulers[dataEngine].Cron(dataIntegrityCronJobs[dataEngine]).Do(m.checkSnapshots, dataEngine)
 
-			if err != nil {
-				return errors.Wrap(err, "failed to schedule snapshot check job")
-			}
-			m.scheduledJobs[dataEngine] = job
+		if err != nil {
+			return errors.Wrap(err, "failed to schedule snapshot check job")
+		}
+		m.scheduledJobs[dataEngine] = job
```

Comment:
I just noticed that this map is not protected by the lock. I am not sure what the scope of that lock is. Anyway, it's another topic